index.js

```diff
@@ -204,6 +224,37 @@ function loader (registryOrVersion) {
       }
       throw new Error("Don't know how to get spawn egg mob name for this mc version")
     }
+
+    // TODO: test this
+    canPlaceOn (block) {
+      if (!this.nbt || !this.nbt.value.CanPlaceOn) return true
```

Comment:
Is this actually a thing?

Reply:
according to the wiki it is, but like I said I need to test it

Reply:
Link?

Reply:
also probably not a thing on every version

Reply:
https://minecraft.fandom.com/wiki/Tutorials/Command_NBT_tags#Items
now that I look back, CanPlaceOn isn't there, only CanDestroy

Reply:
https://minecraft.fandom.com/wiki/Tutorials/Command_NBT_tags#Blocks
It only exists for blocks, which makes sense as only blocks can be placed

Reply:
Ok, then if it's like this for Java Edition it is most likely stored this way on Bedrock Edition also on disk. In which case the canPlaceOn/canDestroy should not be fields on Item instance. They should retrieve and store data in NBT like enchantments.

Reply:
So handle them the same way enchantments are handled?